src/layman/layer/rest_workspace_layers.py

```diff
@@ -129,10 +129,13 @@ def post(workspace):
     name_input_file_by_layer = time_regex is None or input_files.is_one_archive
     enable_more_main_files = time_regex is not None
 
+    # register layer uuid
+    uuid_str = register_publication_uuid_to_redis(workspace, LAYER_TYPE, layername, input_uuid)
```

Comment:
We do not unregister uuid if publication posting fails.

Reply:
Unregistration implemented